pkg/azure/provider.go

```diff
@@ -281,20 +309,36 @@ func (p *Provider) MountSecretsStoreObjectContent(ctx context.Context, attrib ma
 		usePodIdentity = true
 	}
 
+	useVMManagedIdentity := false
+	if useVMManagedIdentityStr == "true" {
+		useVMManagedIdentity = true
+	}
+
+	if usePodIdentity && useVMManagedIdentity {
+		return fmt.Errorf("cannot enable both pod identity and assigned user identity")
+	}
+
 	log.Infof("mounting secrets store object content for %s/%s", p.PodNamespace, p.PodName)
-	if !usePodIdentity {
-		log.Infof("not using pod identity to access keyvault")
+	if !usePodIdentity && !useVMManagedIdentity {
+		log.Infof("not using pod identity or vm assigned user identity to access keyvault")
 		p.AADClientID, p.AADClientSecret, err = GetCredential(secrets)
 		if err != nil {
 			log.Infof("missing client credential to access keyvault")
 			return err
 		}
-	} else {
+	}
+	if usePodIdentity {
 		log.Infof("using pod identity to access keyvault")
 		if p.PodName == "" || p.PodNamespace == "" {
 			return fmt.Errorf("pod information is not available. deploy a CSIDriver object to set podInfoOnMount")
 		}
+	} else if useVMManagedIdentity {
```

Comment:
why do we need this check? `podName` and `podNamespace` are required for pod identity mode only